sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -282,6 +285,27 @@ def _test_py_compile_yaml(self, file_base_name):
     finally:
       shutil.rmtree(tmpdir)
 
+  def _test_sample_py_compile_yaml(self, file_base_name):
```

Comment:
Since the directory name does not have to be the same as the file name, maybe we should have this function expose two params: sample_dir_name and file_base_name. WDYT?

Reply:
That's a good point. Do you project that there'll be cases where, for example, we have multiple .py samples under the same dir demonstrating different facets of a functionality? If so, do you prefer defining a test case to catch all of them or defining separate cases?

Reply:
I discussed the multi-layer directory structure with Jessie and decides that we will adopt the current design where each directory in the core contains one sample, which might include some dependent files. And, the dependent files can also be a python file, however, there is only one main entry python file.

Reply:
I see. So the status quo is that each dir contains one .py file that needs to be picked by unit test. Then I suggest to keep both names consistent in favor of sustainability. WDYT?

Reply:
Agreed. In the efforts of reorganizing the samples, we need to sync with the kubeflow/website to make sure the links there also work.

Reply:
Updated #1750

Reply:
Thanks

Reply:
BTW, regardless of whether the sample dirname is the same as the sample filename, it might still be good practice for this function to expose both the dirname and filename such that it is generic enough to accommodate future changes.  WDYT?